storage_service/administration/validators.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import absolute_import
+import six
+
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext as _
+
+
+class PasswordComplexityValidator(object):
+    """Custom password complexity validator.
+
+    To pass validation, passwords must contain at least three of:
+    - uppercase characters
+    - lowercase characters
+    - numbers
+    - special characters
+    """
+
+    HELP_TEXT = _(
+        "Your password must contain at least 3 of: uppercase characters, "
+        "lowercase characters, numbers, and special characters (symbols or "
+        "non-alphanumeric Unicode characters)."
+    )
```

Comment:
This string would be fine with `black` on a single line. I prefer to do it like that nowadays because it avoids having to ensure a string terminates with a space, or begins in a space depending on who is encoding the string, i.e. written as a single stream, it tends to be self correcting. Not a huge deal if you prefer it this way. It's just me asking _"wait, did `black` format this like that?"_

This reminds me though - you'll need to make translations now you're adding this nice message! 

(The message looks good to me, even though it's verbose - and I tried a password with emoji and it worked!! :sweat_smile:)

Reply:
> It's just me asking "wait, did black format this like that?"

Sharp eye - `black` in fact, did not format this for me! I'll keep that in mind about letting the formatter do the work moving forward. If we're using it, may as well rely on it!

> This reminds me though - you'll need to make translations now you're adding this nice message!

Good point! I'll do that as the last step before merging. Thanks for the reminder! :smile: 

> (The message looks good to me, even though it's verbose - and I tried a password with emoji and it worked!!)

That's great to hear! I briefly flirted with whether it would make sense to normalize as well, but figure that since the password is being hashed anyway, as long as the user is consistent about how they enter the password, it should work out in the end. I suspect we won't have too many users trying to put emoji in their passwords anyway, but then again who knows!